.github/workflows/continuous-integration-workflow.yml

```diff
@@ -39,7 +39,8 @@ jobs:
             backend: 'OPENMP'
           - distro: 'ubuntu:intel'
             cxx: 'icpx'
-            cxx_extra_flags: '-fp-model=precise -Wno-pass-failed'
+            cxx_extra_flags: '-fp-model=precise -Wno-pass-failed -fsanitize=address,undefined -fno-sanitize=function -fno-sanitize-recover=all'
```

Comment:
What are these `-fno-sanitize=function -fno-sanitize-recover=all` flags?

Reply:
We get more warnings when checking function signatures, mostly due to our tools interface since C- and C++- function pointers are technically different types. This pull request fixes a bunch of them but I didn't see a way to address all of them and rather decided to disable this particular check. By default the undefined behavior sanitizer doesn't error out but just warns. `-fno-sanitize-recover=all` makes sure that we actually error out.